### ISSUE TITLE
I am always getting lost

### ISSUE THREAD

**Issue by petpipe123** [aspnet/Docs#10805]
This document has interwoven two different approaches and it is very hard to follow. My suggestion is to have a document that uses the older approach of using Client Id & Secret with KeyVault and *a different document* using MSI. Code snippets have no context after you get halfway through this existing document.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 2cc0b77a-df21-7007-5cec-4399d98c6dc6
* Version Independent ID: efe5ae5e-ff64-f1c7-6920-4512fc8f36d1
* Content: [Azure Key Vault Configuration Provider in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/security/key-vault-configuration?view=aspnetcore-2.2)
* Content Source: [aspnetcore/security/key-vault-configuration.md](https://github.com/aspnet/Docs/blob/master/aspnetcore/security/key-vault-configuration.md)
* Product: **aspnet-core**
* GitHub Login: @guardrex
* Microsoft Alias: **riande**

**Comment by guardrex** [comment]
Hello @petpipe123 ... Thanks for commenting.

The recommendation is to adopt either deployment approach, and everything for the two approaches is covered by their respective sections ...

* Use an external (non-Azure) deployment with AppID and Password (Client Secret) covered by this section :point_right: https://docs.microsoft.com/aspnet/core/security/key-vault-configuration#use-application-id-and-client-secret
* Use an Azure-based deployment with MSI covered by this section :point_right: https://docs.microsoft.com/aspnet/core/security/key-vault-configuration#use-the-managed-service-identity-msi-provider

Nothing is interwoven. Except for the code shown in those two sections, everything else in the topic applies equally to both approaches.

If you set the preprocessor directive at the top of the *Program.cs* file to `Basic`, you'll see VS light up the AppID/Password/external app code in the file. When that directive is set to `Managed`, VS lights up the MSI-based code.

We could split, but it would create a huge amount of content duplication.

I think what I need to do is ...

1. At the top explain the two approaches in a few lines ... the choice of how to deploy and what consequence it has for using either the AppID/Passoword or MSI sections.
1. Perhaps change those section headings to make them clearer.

**Comment by petpipe123** [comment]
Hi Luke,

The “In this article” right sidebar provides great navigation, but within the documentation, it is not intrinsic in the reader’s mind to understand and pay attention to “logical page breaks”, i.e. when content spans beyond the visible window.

Would it be “cheesy” to have the document highlighted text when it corresponds to the section on the right sidebar providing visual context?  To provide you an example, using the VS IDE has great visual context of where “you” are within the codebase.

Cheers,

Peter

**Comment by guardrex** [comment]
> highlighted text when it corresponds to the section on the right sidebar providing visual context

It's beyond our ability to provide that here because we mostly manage the ASP.NET Core content, but that's great feedback for the Microsoft Docs repo, where the managers receive suggestions and feedback. Make that request at https://github.com/MicrosoftDocs/feedback/issues.
